### PR TITLE
Run Codemod tests in CI pipeline

### PR DESCRIPTION
**Pull request by KaroAug**
## Changes

Run Codemod tests in CI pipeline.

## Testing

N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -55,3 +55,7 @@ test-results
 common/api/summary/summary.exports.csv
 
 debug.log
+
+# Codemod specific
+test
```
Comment (GerardasB) [inline .gitignore:60]: This will ignore all test files together with junit report.

File: common/config/azure-pipelines/templates/core-build.yaml [hunk 2]
```diff
@@ -86,6 +86,12 @@ steps:
       CI: true
     condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
 
+  # Run codemod tests
+  - script: npm run test
+    displayName: run Codemod Tests
+    workingDirectory: "tools/codemod"
+    condition: and(succeeded(), ne(variables['Agent.OS'], 'Linux'))
```
Comment (GerardasB) [inline common/config/azure-pipelines/templates/core-build.yaml:93]: Why is this condition needed?

File: tools/codemod/jest.config.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
 module.exports = {
   transform: { "\\.ts$": ["ts-jest"] },
+  reporters: [
+    "default",
+    [
+      "jest-junit",
+      {
```
Comment (GerardasB) [inline tools/codemod/jest.config.js:7]: Maybe let's just keep defaults and gitignore `junit.xml`?

**Review by GerardasB — CHANGES_REQUESTED** [review]

File: common/config/azure-pipelines/templates/core-build.yaml (hunk 2 above)
Comment (KaroAug) [reply common/config/azure-pipelines/templates/core-build.yaml:93]: Due to unexpected failures only in Linux environment

Comment (GerardasB) [reply common/config/azure-pipelines/templates/core-build.yaml:93]: Ok let's keep this for now, need to determine why `jscodeshift` extension methods https://github.com/facebook/jscodeshift?tab=readme-ov-file#extensibility are not working on Linux.

File: tools/codemod/jest.config.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
   transform: { "\\.ts$": ["ts-jest"] },
+  reporters: ["default", ["jest-junit", {}]],
```
Comment (GerardasB) [inline tools/codemod/jest.config.js:3]:
```suggestion
  reporters: ["default", "jest-junit"],
```

**Review by GerardasB — APPROVED** [review]

**Pull request merged**